crates/rattler_installs_packages/src/wheel_builder/build_environment.rs

```diff
@@ -280,7 +327,7 @@ impl<'db> BuildEnvironment<'db> {
         ));
 
         Ok(BuildEnvironment {
-            work_dir,
+            work_dir: RwLock::new(DeleteOrPersist::Delete(Some(work_dir))),
```

Comment:
I think you are missing the point of the type :), it's more like the state it's in than the actual operation, the fact that it is a persist type means that the directory will be persisted, because `into_path` has been called. So the `if` in the code you referenced is to see if we need to call the `persist` function, so if the type is in the `Persist` state it means it will already be persisted.

Reply:
🤦  sorry I got it now

Reply:
sorry I got it now 🤦